### ISSUE TITLE
LXC configuration issues on Ubuntu 18.04 (lxc 3.0.0)

### ISSUE THREAD

**Issue by sgalgano** [adjacentlink/emane-tutorial#16]
Reported on mailing list by Michael J. Vincent:

 I've compiled the latest 1.2.2 and Protobuf 3.5.1; however, none of the EMANE tutorial simulations (https://github.com/adjacentlink/emane-tutorial) work.  Investigating the issues why nothing starts - it seems LXC is at major version 3.x on Ubuntu 18.04 and that means config syntax changes (https://github.com/lxc/lxd/issues/4396#issuecomment-378322166).

I updated the scripts/demctl-init.sh script to accommodate; however, I'm still having issues - now with OLSRd not finding the /proc filesystem on the containers.

**Comment by vinsworldcom** [comment]
Steve,
Thanks for opening the issue.  Attached are the patch file for democtl-host that looks at LXC version and then use 2 new files for lxc2 and lxc3 templates to create the lxc.conf files.

[emane-tutorial.zip](https://github.com/adjacentlink/emane-tutorial/files/2027097/emane-tutorial.zip)

**Comment by vinsworldcom** [comment]
The OLSRd finding /proc filesystem is fixed as well in the lxc3.template file in the ZIP patch above.  Essentially need the command:
lxc.mount.auto = cgroup:mixed proc:mixed sys:mixed

**Comment by sgalgano** [comment]
I'm going to stick with the modifications that were made in `feature/issue16`.

We have on our roadmap a revamped version of the tutorial that drops the use of makefiles and sed in favor of a template library. This will likely either use [etce](https://github.com/adjacentlink/python-etce) or an *etce-like* offshoot. We have largely been using [Mako](http://www.makotemplates.org/) as our template language on other projects.

I've tested `feature/issue16` modifications on CentOS 7, Fedora 27 and Ubuntu 18.04. The feature branch has now been merged into `develop`.
